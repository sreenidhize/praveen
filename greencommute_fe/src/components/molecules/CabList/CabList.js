import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
    root:{
        paddingTop:"10px",
        marginTop:"10px"
    },

    title: {
        color: theme.typography.h3.color,
        textAlign: "left",
        paddingLeft:"10px",
    },
    fare: {
        textAlign: "left",
        paddingLeft:"10px"
    },
    book: {
        color: theme.palette.primary.main,  
        cursor: "pointer"
    },
    img: {
        marginRight: "10px",
        height: "42px",
        width: "42px",
    },
}));

const CabList = (props) => {
    const classes = useStyles();

    return (
        <Grid container spacing={1} direction="column" justify="center" 
        className={classes.root} data-testid="cablist-placeholder">
           {options.map((item) => {
                return (
                    <Grid item>
                    <Grid container direction="row" alignItems="center" key={item.type}>
                        <img className={classes.img} src={item.url} alt={item.type}></img>
                        <Grid item container xs={7} direction="column" justify="center">
                            <Typography variant="h5" className={classes.title}>{item.type}</Typography>
                            <Typography variant="body2" className={classes.fare}>Approx {item.value}</Typography>
                        </Grid>
                         <Typography variant="h5" className={classes.book}>Book now</Typography>
                    </Grid>
                    </Grid>
                );
            })}

       </Grid>
    );
}
export const options = [
    { 
        type: "Ola", 
        value: "$40", 
        url: "https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg" 
    },
    { 
        type: "Uber", 
        value: "$40", 
        url: "https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg" 
    }
]

CabList.propTypes = {
    options: PropTypes.array,
}

export default CabList;