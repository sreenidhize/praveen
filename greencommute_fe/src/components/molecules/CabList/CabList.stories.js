import React from 'react';
import CabList from './CabList';

export default {
    title: 'molecules/CabList',
    component: CabList,
};

const Template = ({...args}) => <div style={{width:"400px"}}><CabList {...args} /></div>

export const Primary = Template.bind({});
Primary.args = {
};
