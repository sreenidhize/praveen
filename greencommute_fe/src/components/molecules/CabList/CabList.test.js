import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CabList from './CabList';
import options from './CabList';

describe("CabList test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <CabList options={options} />
      );
      expect(container).toMatchSnapshot();
    });
  
    test("CabList render correctly", () => {
      const { getByTestId } = render(<CabList options={options} />);
      expect(getByTestId("cablist-placeholder")).toBeInTheDocument();
    });
});