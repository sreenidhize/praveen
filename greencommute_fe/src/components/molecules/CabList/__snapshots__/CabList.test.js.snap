// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CabList test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
    data-testid="cablist-placeholder"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <img
          alt="Ola"
          class="makeStyles-img-5"
          src="https://res-5.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco/v1459446250/xx0u5xwwzygivm67ogsk.jpg"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-7"
        >
          <h5
            class="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
          >
            Ola
          </h5>
          <p
            class="MuiTypography-root makeStyles-fare-3 MuiTypography-body2"
          >
            Approx 
            $40
          </p>
        </div>
        <h5
          class="MuiTypography-root makeStyles-book-4 MuiTypography-h5"
        >
          Book now
        </h5>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      >
        <img
          alt="Uber"
          class="makeStyles-img-5"
          src="https://semaine-production.s3.amazonaws.com/products/product/image/6827/large_Uber-app.jpg"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-xs-center MuiGrid-grid-xs-7"
        >
          <h5
            class="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
          >
            Uber
          </h5>
          <p
            class="MuiTypography-root makeStyles-fare-3 MuiTypography-body2"
          >
            Approx 
            $40
          </p>
        </div>
        <h5
          class="MuiTypography-root makeStyles-book-4 MuiTypography-h5"
        >
          Book now
        </h5>
      </div>
    </div>
  </div>
</div>
`;
